src/mui/input/ReferenceInput.js

```diff
@@ -108,10 +108,8 @@ export class ReferenceInput extends Component {
     }
 
     setFilter = (filter) => {
-        if (filter !== this.params.filter) {
```

Comment:
I think you still need to keep the previous `searchText` and compare it with the new one to avoid needless fetches.